### PR TITLE
strictly confined snap

### PR DESCRIPTION
**Pull request by marcusboden**
This is my first draft of a strictly confined tailscale snap. I've tested a number of features (general connectivity, `tailscale ssh`, `tailscale netcheck` and some more) and will fix other issues as I run into them.
Also bumped to a more recent version and to core22.

Any feedback is welcome!

### CODE REVIEW

File: snapcraft.yaml [hunk 1]
```diff
@@ -14,23 +14,48 @@ architectures:
   - build-on: arm64
   - build-on: armhf
 
+plugs:
+  sys-devices-virtual-info:
```
Comment (sergiusens) [inline snapcraft.yaml:18]: there's a policy of naming this like the path being granted access to, might want to double check with security if this is the right name

File: snapcraft.yaml [hunk 2]
```diff
@@ -14,23 +14,48 @@ architectures:
...
+    build-snaps:
+      - go
+    stage-packages:
+      - iptables
+      - openssh-client # for tailscale ssh
     override-build: |
       sed -i paths/paths.go -e 's|"/var/run/tailscale/tailscaled.sock"|filepath.Join(os.Getenv("SNAP_DATA"), "tailscaled.sock")|'
-      snapcraftctl build
+      go install -p $CRAFT_PARALLEL_BUILD_COUNT ./cmd/tailscale
+      go install -p $CRAFT_PARALLEL_BUILD_COUNT ./cmd/tailscaled
```
Comment (sergiusens) [inline snapcraft.yaml:57]: why is this needed now?

Comment (marcusboden) [reply snapcraft.yaml:57]: Because we don't need to build all possible binaries. Those two are the ones that are shipped /w other packages as well.

File: snapcraft.yaml (hunk 1 above)
Comment (marcusboden) [reply snapcraft.yaml:18]: I've only found [this](https://snapcraft.io/docs/system-files-interface):
> Note, when declaring an instance of the system-files plug as above, it should be named with a descriptive name that indicates to a user what access it grants. In this case, the name etc-foo is used to reflect the access to /etc/foo.

I thought that name is descriptive enough. But if not, I can also change it to `sys-devices-virtual-dmi-id`.

**Comment by marcusboden** [comment]
Are there any more changes I should make?

**Review by sergiusens — APPROVED** [review]

**Comment by sergiusens** [comment]
thanks and sorry for the delay
